### PR TITLE
Updated Tutorial

### PR DESCRIPTION
**Pull request by Gomango999**
Hi all!

I'm Kevin, working <strike>under</strike> with @iosonofabio, and tasked with writing new tutorials for python-igraph. The new tutorials are targetted towards more concise, bite-sized articles which each present self-contained ideas, with a greater focus on fast and usable code examples. As a first commit, I've added a "quick start" tutorial, which includes ~50 lines of code for an individual who is already familiar with both python and graphs, who want a quick and dirty way to get started with igraph.

Please don't merge this yet, this is still a placeholder and I plan to add more to the list of tutorials. I'll let you all know when I need a review.

### CODE REVIEW

File: doc/source/tutorial/assets/america.gml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Creator "igraph version 0.9.3 Sun Nov 14 18:16:32 2021"
+Creator "igraph version 0.10.0-dev+63a49e74 Sun Nov 21 16:02:47 2021"
```
Comment (ntamas) [inline doc/source/tutorial/assets/america.gml:1]: It would be better to keep a non-dev version number in any file that is included in the tutorial.

Comment (Gomango999) [reply doc/source/tutorial/assets/america.gml]: Good catch. I'm not actually using this file, so I've removed it for now, but definitely a good thing to keep in mind for the future.

File: doc/source/tutorials/bipartite_matching/assets/maxflow2.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+fig, ax = plt.subplots()
+ig.plot(
+    g,
+    target=ax,
+    layout=layout,
+    vertex_size=0.4,
+    vertex_label=range(g.vcount()),
+    vertex_color=["lightblue" if i < 9 else "orange" for i in range(11)],
+    edge_width=[1.0 + flow.flow[i] for i in range(g.ecount())]
```
Comment (szhorvat) [inline doc/source/tutorials/bipartite_matching/assets/maxflow2.py:37]: What version are you using? This line does not seem to work with 0.9.8. It wants a single number, not a list.

File: doc/source/tutorials/bipartite_matching/assets/maxflow2.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+for i in range(4, 9):
+    g.vs[i]["type"] = False
+
+# Add source and sink as nodes 9 and 10
+g.add_vertices(2)
+g.add_edges([(9, 0), (9, 1), (9, 2), (9, 3)]) # connect source to one side
+g.add_edges([(4, 10), (5, 10), (6, 10), (7, 10), (8, 10)]) # ... and sinks to the other
+
+flow = g.maxflow(9, 10)
+print("Maximal Matching is:", flow.value)
```
Comment (szhorvat) [inline doc/source/tutorials/bipartite_matching/assets/maxflow2.py:22]: Let's be precise: this is not "the matching". It is the _size_ of the matching. The matching consists of a set of edges. It is not a number.

File: doc/source/tutorials/bipartite_matching/bipartite_matching.rst [hunk 4]
```diff
@@ -0,0 +1,60 @@
+==========================
+Maximal Bipartite Matching
```
Comment (szhorvat) [inline doc/source/tutorials/bipartite_matching/bipartite_matching.rst:2]: "Maximum", not "maximal". It's not the same.

If we discuss maximum bipartite matchings, it's important to point out that there is already a function for this. This is a nice example, but people should not get the impression that they need to resort to flow calculations to get a bipartite matching. It would be nice to compare with the result of `.maximum_bipartite_matching()`.

File: doc/source/tutorials/quickstart/quickstart.rst [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    import igraph as ig
+    import matplotlib.pyplot as plt
+
+    # Construct a graph with 3 vertices
+    n_vertices = 3
+    edges = [(0, 1), (1, 2), (2, 0)]
+    g = ig.Graph(n_vertices, edges)
+
+    # Set attributes for the graph, nodes, and edges
+    g["title"] = "Cities of America"
```
Comment (szhorvat) [inline doc/source/tutorials/quickstart/quickstart.rst:17]: This will be an absolutely subjective nitpick and I might get shot down, but here it goes ... I thin the English-speaking internet is much too US-centric as it is. People use American examples without thinking all the time even when they are not Americans and have never been there. If you are American and particularly love this example, then please go ahead. Otherwise, can we choose a neutral example please?

File: doc/source/tutorials/bipartite_matching/assets/maxflow2.py (hunk 2 above)
Comment (Gomango999) [reply doc/source/tutorials/bipartite_matching/assets/maxflow2.py]: I'm using the latest develop branch. There are a few reasons why I'm doing this:

- Fabio and I have agreed that we will be using Matplotlib over Cairo, since Cairo is being phased out.
- Matplotlib on master is lacking a few features, such as setting each edge widths individually shown above, and the need to manually remove the spines and ticks from the output file, which would unecessarily complicate the code examples.
- Since the tutorial is not coming out immediately, it makes sense to be able to use features from develop. 

Of course, the downside is that the develop branch is susceptible to change, invalidating parts of the tutorial. If you think that there's going to be alot of changes to the matplotlib drawer or have other concerns, I'm happy to rework this to use the main version.

File: doc/source/tutorials/bipartite_matching/assets/maxflow2.py (hunk 3 above)
Comment (Gomango999) [reply doc/source/tutorials/bipartite_matching/assets/maxflow2.py]: Fixed.

File: doc/source/tutorials/bipartite_matching/bipartite_matching.rst (hunk 4 above)
Comment (Gomango999) [reply doc/source/tutorials/bipartite_matching/bipartite_matching.rst]: My bad, I missed that particular method while going through the API. I've modified the example to use `.maximum_bipartite_matching()` and moved the original tutorial to the `archived/` folder. 

I would like these examples to show the most efficient / standard way of doing things, so I don't think that calculating the matching by manually making a max flow graph belongs here.

File: doc/source/tutorials/quickstart/quickstart.rst (hunk 5 above)
Comment (Gomango999) [reply doc/source/tutorials/quickstart/quickstart.rst]: Your assessment was spot on! I'm not American ( born in Australia :) ) so it also surprises me that I went to a American example without thinking about it. 

I've changed the example to a social network to make it more neutral.

File: doc/source/tutorials/bipartite_matching/bipartite_matching.rst (hunk 4 above)
Comment (szhorvat) [reply doc/source/tutorials/bipartite_matching/bipartite_matching.rst]: I do think it was a very good and educational example, and I would very much suggest keeping it. Just make it clear that this is a not the most efficient way of finding a maximum bipartite matching.

As I see it, this example is not about a specific task (bipartite matching). Instead, it demonstrates how to work with igraph in general, and gives you the opportunity to show several basic operations, such as adding edges, setting up a bipartite graph, retrieving results from a flow object, highlighting a subset of edges, etc. For this, it's an excellent example.

Comment (szhorvat) [reply doc/source/tutorials/bipartite_matching/bipartite_matching.rst]: In fact, I would encourage you to try to show off even more functionality within the context of this example, such as `is_biparite`, `Graph.Bipartite`, or whatever you think fits. It's nice to show more than one way of doing things, such as setting the `type` property directly or through `Graph.Bipartite`.

Comment (Gomango999) [reply doc/source/tutorials/bipartite_matching/bipartite_matching.rst]: That makes complete sense, and I'll let my supervisor @iosonofabio make the final call. He's currently on break, so I'll leave this in the archives for now.

Comment (iosonofabio) [reply doc/source/tutorials/bipartite_matching/bipartite_matching.rst]: Happy to go with whatever @szhorvat proposed, even if it makes it a little longer

**Comment by szhorvat** [comment]
It would be nice to add examples showing how to visualize node or edge properties such as various centralities. For example, visualize both vertex betweenness and edge betweenness in the same graph using colour, picking an arbitrary colour scheme.

**Comment by Gomango999** [comment]
That sounds like an excellent idea! I'll get right on it after I return from exams :)

**Comment by iosonofabio** [comment]
Thank you @Gomango999 for the PR.

I've just committed c2a4843 which centers vertex labels horizontally and vertically, which makes them look a little more tidy.

It would be great to have a tutorial where edges are plotted with both custom widths and colors, using `edge_width=` and `edge_color=`. At the moment there might be a bug with the color that I'm planning to fix later today.

When you're back from exams you could merge develop into this branch, remake the figures with the centered labels, and add variable edge color to one of the tutorials. Then we could open this up for other devs' comments

File: doc/source/tutorials/quickstart/assets/quickstart.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    target=ax,
+    layout="circle", # print nodes in a circular layout
+    vertex_size=0.1,
+    vertex_color=["lightblue" if gender == "M" else "pink" for gender in g.vs["gender"]],
+    vertex_frame_width=2.0,
+    vertex_frame_color="white",
+    vertex_label=g.vs["name"],
+    vertex_label_size=7.0,
+    edge_width=[2 if married else 1 for married in g.es["married"]],
+)
```
Comment (iosonofabio) [inline doc/source/tutorials/quickstart/assets/quickstart.py:33]: perhaps add a line:

```python
    edge_color=["black" if married else "red" for married in g.es["married"]],
```

to showcase that option

**Comment by iosonofabio** [comment]
Alright, I actually went out and fixed those things and all plot aspect ratios, which were a little off.

I also merged from develop which includes the fix to center horizontally and vertically vertex labels.

I also made a "gallery" page, added entries for gallery and all tutorials to the sphinx TOC tree, added global refs and includes, and cross-linked a little.

This should be ready for a first review now.

**Comment by h5jam** [comment]
@Gomango999 Hi, Kevin! I'm Seungoh Han. I was thinking of adding examples of topological sort to this PR. I would like to work on this if it's possible. Can I join your PR so that I can work on this job?

**Comment by iosonofabio** [comment]
Please make a separate PR

File: doc/source/tutorials/bipartite_matching/assets/bipartite_matching.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+assert(g.is_bipartite())
+
+matching = g.maximum_bipartite_matching()
+
+# Print pairings for each node on one side
+matching_size = 0
+print("Matching is:")
+for i in range(5):
+    print(f"{i} - {matching.match_of(i)}")
+    if matching.match_of(i):
```
Comment (ntamas) [inline doc/source/tutorials/bipartite_matching/assets/bipartite_matching.py:18]: This happens to work in the current case because `matching.match_of()` never returns 0. However, if you were running this on `for i in range(5, 10)`, the size of the matching would be off by one because `matching.match_of(5)` is 0, and 0 is falsy in Python.

it is safer to use `matching.match_of(i) is not None` or to call `matching.match_of()` with a `Vertex` object (i.e. `g.vs[i]`, in which case `matching.match_of()` returns a `Vertex` object, which is never falsy.

File: doc/source/tutorials/bipartite_matching/assets/bipartite_matching.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
+import igraph as ig
+import matplotlib.pyplot as plt
+
+# Assign nodes 0-4 to one side, and the nodes 5-8 to the other side
+g = ig.Graph.Bipartite(
+    [0, 0, 0, 0, 0, 1, 1, 1, 1],
+    [(0, 5), (1, 6), (1, 7), (2, 5), (2, 8), (3, 6), (4, 5), (4, 6)]
+)
+assert(g.is_bipartite())
```
Comment (ntamas) [inline doc/source/tutorials/bipartite_matching/assets/bipartite_matching.py:9]: It is a bad practice to use parentheses around an `assert` call because assert is a _statement_ in Python, not a function. `assert(x)` happens to be okay, but `assert(x, failure_message)` is parsed as `assert (x, failure_message)`, where `(x, failure_message)` is treated as a tuple of length 2, which is always true. In other words:

```python
>>> assert False, "this assertion is False"    # expected to fail
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: this assertion is False
>>> assert(False, "this assertion is False")   # will not fail because it is interpreted as a tuple
```

TL;DR: please remove the parentheses

File: doc/source/tutorials/bipartite_matching/assets/bipartite_matching.py [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+fig, ax = plt.subplots(figsize=(7, 3))
+ig.plot(
+    g,
+    target=ax,
+    layout=g.layout_bipartite(),
+    vertex_size=0.4,
+    vertex_label=range(g.vcount()),
+    vertex_color="lightblue",
+    edge_width=[2.5 if e.target==matching.match_of(e.source) else 1.0 for e in g.es]
```
Comment (ntamas) [inline doc/source/tutorials/bipartite_matching/assets/bipartite_matching.py:30]: Use spaces around `==`.

File: doc/source/tutorials/bipartite_matching_maxflow/assets/bipartite_matching_maxflow.py [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+g = ig.Graph(
+    9,
+    [(0, 4), (0, 5), (1, 4), (1, 6), (1, 7), (2, 5), (2, 7), (2, 8), (3, 6), (3, 7)],
+    directed=True
+)
+
+# Assign nodes 0-3 to one side, and the nodes 4-8 to the other side
+for i in range(4):
+    g.vs[i]["type"] = True
```
Comment (ntamas) [inline doc/source/tutorials/bipartite_matching_maxflow/assets/bipartite_matching_maxflow.py:12]: `g.vs[range(4)] = True` also works; not sure if you actually want to show that or just want to be explicit about the looping, both are okay for me

File: doc/source/tutorials/bipartite_matching_maxflow/assets/bipartite_matching_maxflow.py [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+for i in range(4, 9):
+    g.vs[i]["type"] = False
+
+# Add source and sink as nodes 9 and 10
+g.add_vertices(2)
+g.add_edges([(9, 0), (9, 1), (9, 2), (9, 3)]) # connect source to one side
+g.add_edges([(4, 10), (5, 10), (6, 10), (7, 10), (8, 10)]) # ... and sinks to the other
+
+flow = g.maxflow(9, 10)
+print("Size of Maximum Matching (maxflow) is:", flow.value)
```
Comment (ntamas) [inline doc/source/tutorials/bipartite_matching_maxflow/assets/bipartite_matching_maxflow.py:22]: Please capitalize this as `maximum matching` as it not a proper noun

File: doc/source/tutorials/bipartite_matching_maxflow/assets/bipartite_matching_maxflow.py [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+
+# Compare this to the "maximum_bipartite_matching()" function
+g2 = g.copy()
+g2.delete_vertices([9, 10]) # delete the source and sink, which are unneeded for this function.
+matching = g2.maximum_bipartite_matching()
+
+matching_size = 0
+for i in range(4):
+    if matching.match_of(i):
+        matching_size += 1
```
Comment (ntamas) [inline doc/source/tutorials/bipartite_matching_maxflow/assets/bipartite_matching_maxflow.py:32]: Probably a more pythonic way to say this is `matching_size = sum(1 for i in range(4) if matching.match_of(i) is not None)`. Again, watch out for the potential case of `matching.match_of(i) == 0`.

File: doc/source/tutorials/bipartite_matching_maxflow/bipartite_matching_maxflow.rst [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+
+.. tutorials-bipartite-matching-maxflow
+
+==========================================
+Maximum Bipartite Matching by Maximum Flow
+==========================================
+
+This example presents how to visualise bipartite matching using maximum flow. Please note that the *igraph* already has :meth:`maximum_bipartite_matching` which is better suited for finding the maximum bipartite matching. For a demonstration on how to use that method instead, check out `Maximum Bipartite Matching<>`_. This particular example is purely for demonstrative purposes.
+
+.. TODO: add link to Maximum Bipartite Matching
```
Comment (ntamas) [inline doc/source/tutorials/bipartite_matching_maxflow/bipartite_matching_maxflow.rst:11]: Remove the TODO comment if the link is already there (it seems like it is)

File: doc/source/tutorials/bipartite_matching_maxflow/bipartite_matching_maxflow.rst [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+
+    # Generate the graph
+    g = ig.Graph(
+        9,
+        [(0, 4), (0, 5), (1, 4), (1, 6), (1, 7), (2, 5), (2, 7), (2, 8), (3, 6), (3, 7)],
+        directed=True
+    )
+
+    # Assign nodes 0-3 to one side, and the nodes 4-8 to the other side
+    for i in range(4):
```
Comment (ntamas) [inline doc/source/tutorials/bipartite_matching_maxflow/bipartite_matching_maxflow.rst:26]: Another way of saying this: `g.vs["type"] = [vertex.index < 4 for vertex in g.vs]`; again, I'm okay with an explicit for loop but just wanted to make sure that you are aware of this.

File: doc/source/tutorials/quickstart/quickstart.rst [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+
+.. tutorials-quickstart
+
+===========
+Quick Start
+===========
+
+This quick start will demonstrate the following:
+
+- Construct a *igraph* graph from scratch
```
Comment (ntamas) [inline doc/source/tutorials/quickstart/quickstart.rst:11]: I think it should be `an *igraph* graph` (note the `an` vs `a`)

File: doc/source/tutorials/shortest_paths/assets/shortest_path.py [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+g.es["weight"] = [2, 1, 5, 4, 7, 3, 2]
+
+# g.get_shortest_paths() returns a list of edge ID paths
+results = g.get_shortest_paths(0, to=5, weights=g.es["weight"], output="epath") # results = [[1, 3, 5]]
+
+if len(results[0]) > 0:
+    # Add up the weights across all edges on the shortest path
+    distance = 0
+    for e in results[0]:
+        distance += g.es[e]["weight"]
```
Comment (ntamas) [inline doc/source/tutorials/shortest_paths/assets/shortest_path.py:29]: This `for` loop could be made more idiomatic Python as follows: `distance = sum(g.es[e]["weight"] for e in results[0])`.

**Comment by ntamas** [comment]
Thank you guys, it's looking very good! I've proposed suggestions for a few places where I've felt that the examples are speaking Python with a C-ish accent :)

File: doc/source/tutorials/bipartite_matching_maxflow/bipartite_matching_maxflow.rst (hunk 13 above)
Comment (Gomango999) [reply doc/source/tutorials/bipartite_matching_maxflow/bipartite_matching_maxflow.rst:11]: Currently the link doesn't actually link anywhere which is why I've left that TODO there. I'll add a link when I have a better idea of the site structure.

**Comment by Gomango999** [comment]
Thanks for the very thorough review @ntamas, I see you've picked up on my language of origin haha!. I'll get around to fixing all of these tomorrow, and I'll definitely be a bit more cautious with some of the grammar things in the future.

**Comment by ntamas** [comment]
> Thanks for the very thorough review @ntamas, I see you've picked up on my language of origin haha! I'll get around to fixing all of these tomorrow, and I'll definitely be a bit more cautious with some of the grammar things in the future.

I'm not a native speaker so take my suggestions with a grain of salt :)

**Comment by iosonofabio** [comment]
I've implemented the changes - still assuming you have exams Kevin ;-)

@szhorvat thank you for the link, yes that's quite useful. As you can see from the PNGs of this PR, matplotlib has parity with Cairo as far as I could see, so in my mind nothing obvious needs to be implemented right now. The only thing I did notice is a bug with list of named colors for `edge_color`, I'll take a look ASAP. If you noticed any particular missing features, please bring them to my attention and I'll take a look, otherwise Kevin could take inspiration from that link when he sets up a few more tutorials over the next few weeks

**Comment by iosonofabio** [comment]
@Gomango999 I added the stub of another tutorial that shows matplotlib animations with a ring graph. I think it's useful as a starting for people looking for similar things, and might prompt some interesting feature requests. Would you mind fleshing out the rst file a little?

Thank you!

**Comment by Gomango999** [comment]
> I've implemented the changes - still assuming you have exams Kevin ;-)

Beat me to it, much appreciated :) 

> @Gomango999 I think the main thing that requires fixing now is the hyperlink... wanna give it a go?
> Would you mind fleshing out the rst file a little?

Happy to do both of these! By fixing the hyperlink, did you mean adding working links from the gallery to everywhere else (as well as between the tutorials themselves)? I'm still busy with exams, but these seem simple enough that I can slot some time in somewhere.

**Comment by Gomango999** [comment]
> I wrote up a bunch of small “benchmark tasks” to be able to judge the usability of igraph.
> https://github.com/igraph/igraph/wiki/Collection-of-simple-tasks

@szhorvat I tried using this link, but it brought me back to the main wiki page with a message saying "You do not have permission to update this wiki." Is there some other way I could take a look at these tests?

**Comment by iosonofabio** [comment]
links from the gallery to the tutorials should be there, but links from the tutorials to external sites or other pages of the docs, and from other pages of the docs to the tutorials would be helpful I think?

Good luck with the exams!

**Comment by iosonofabio** [comment]
> > I wrote up a bunch of small “benchmark tasks” to be able to judge the usability of igraph.
> > https://github.com/igraph/igraph/wiki/Collection-of-simple-tasks
> 
> @szhorvat I tried using this link, but it brought me back to the main wiki page with a message saying "You do not have permission to update this wiki." Is there some other way I could take a look at these tests?

The link is broken now, but was working the other day. @szhorvat might know better...

**Comment by ntamas** [comment]
The wiki page mentioned above was renamed and thus moved to https://github.com/igraph/igraph/wiki/Usability-benchmarks

**Comment by ntamas** [comment]
Looks good! I'm putting the PR back into a draft state since Fabio's section still needs to be fleshed out and there are a couple of TODO items left in the .rst files. Let me know when it's ready for another round.

File: doc/source/tutorials/ring_animation/ring_animation.rst [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+
+The received output is:
+
+.. figure:: ./figures/ring_animation.gif
+
+   :alt: The visualisation of a animated ring graph
+   :align: center
+   :caption: Sequentially animated ring graph.
+
+Note that we use *igraph*'s :meth:`Graph.subgraph()` (a.k.a `:meth:`Graph.induced_subgraph()`<https://igraph.org/python/api/0.9.8/igraph._igraph.GraphBase.html#induced_subgraph>`_) in order to obtain a section of the ring graph at a time for each frame.
```
Comment (iosonofabio) [inline doc/source/tutorials/ring_animation/ring_animation.rst:55]: Sphinx (the doc builder) has its own way to emphasize notes:

```rst
.. note::
  We used *igraph*'s :meth:`Graph....
```

Comment (Gomango999) [reply doc/source/tutorials/ring_animation/ring_animation.rst]: Done.

**Comment by Gomango999** [comment]
Hi all, I'm back from exams and ready to tackle another round of tutorial writing :)

> I posted the following in chat, but you may not have noticed it there, so I'll re-post here. It's related to the tutorial building, so @Gomango999 may be interested as well.

These benchmark tests look great! I'd love to work through these and write a tutorial for each one

> links from the gallery to the tutorials should be there, but links [...] from other pages of the docs to the tutorials would be helpful I think?

I've implemented links from the tutorials to the API documentation, but I wasn't sure how to go the other way round. How would I go about linking something (for example max flow) back to the max flow tutorial when I don't know the URL for the tutorial?

**Comment by szhorvat** [comment]
> > I posted the following in chat, but you may not have noticed it there, so I'll re-post here. It's related to the tutorial building, so @Gomango999 may be interested as well.
> 
> These benchmark tests look great! I'd love to work through these and write a tutorial for each one

There is certainly some good tutorial material there, and it makes sense to read through the tasks and take inspiration. But the purpose of this collection of tasks is not to write tutorials, but to judge the usability of the library, partly through comparisons with other libraries. As they say, simple things should be simple to implement, and difficult ones should be possible.

If you are using some of the material for tutorials, I would like to ask you to also contribute it to the (newly created) usability benchmarks repo in a simpler format.

The issue for this is here: #469

**Comment by ntamas** [comment]
> I've implemented links from the tutorials to the API documentation, but I wasn't sure how to go the other way round. How would I go about linking something (for example max flow) back to the max flow tutorial when I don't know the URL for the tutorial?

Unfortunately I think you need to hardcode the URL - this is a disadvantage of the current doc pipeline, which treats the API docs and the tutorial docs as completely separate entities. I'd skip the backlinks from the API to the tutorial for the time being until we find a better solution for this.

**Comment by iosonofabio** [comment]
heya! This is ready for review and possibly merge I think:

- fixed a few things with TOC/hyperlinks
- added the cliques tutorial from the usability benchmark
- extended `mkdoc.sh` to build tutorials in this repo: not visually appealing, but helps making sure all is in place except CSS
- little code improvements here or there

It's ready for prime time AFAICT, guys take a look and we can merge

File: doc/source/tutorials/shortest_paths/shortest_paths.rst [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+    Shortest distance is:  3
+    Shortest weighted distance is:  8    
+
+.. figure:: ./figures/shortest_path.png
+   :alt: The visual representation of a weighted network for finding shortest paths
+   :align: center
+
+   The Graph `g` 
+
+.. TODO: Add in edge weights as labels when possible! Matplotlib does not support displaying edge weights (and the develop branch implementation is bugged).
```
Comment (szhorvat) [inline doc/source/tutorials/shortest_paths/shortest_paths.rst:70]: Wasn't this fixed recently?

File: doc/source/tutorials/erdos_renyi/erdos_renyi.rst [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+    :alt: The visual representation of a randomly generated Erdos Renyi graph
+    :align: center
+
+    Erdos Renyi Random Graphs With Probability ``p`` = 0.2
+
+.. figure:: ./figures/erdos_renyi_m.png
+    :alt: The second visual representation of a randomly generated Erdos Renyi graph
+    :align: center
+
+    Erdos Renyi Random Graphs With Number of Edges ``m`` = 35
```
Comment (szhorvat) [inline doc/source/tutorials/erdos_renyi/erdos_renyi.rst:94]: Here and above: Can you change "Erdos Renyi" to "Erdős-Rényi"?

**Comment by szhorvat** [comment]
Very happy to see this. I left two small comments, but please don't wait for more comments from me before merging.

Can you let me know though how I can build the docs so I can see the final form? I have not done this yet for python-igraph.

**Comment by iosonofabio** [comment]
Thanks @szhorvat - I'll fix those and merge.

@Gomango999 : thank you for the hard work on this, I'll now take over and merge!

**Comment by szhorvat** [comment]
Now that I know how to build the tutorial, I might like to make a few minor improvements (phrasing, clarity, readability of figures). Should I open a PR for these @iosonofabio or just commit directly?

**Comment by Gomango999** [comment]
> thank you for the hard work on this, I'll now take over and merge!

Thanks everyone! I still have some more tutorials in mind (got side tracked a bit these past few days), so these will be coming up in future PRs.
